crates/accelerate/src/convert_2q_block_matrix.rs

```diff
@@ -53,94 +52,147 @@ pub fn get_matrix_from_inst(py: Python, inst: &PackedInstruction) -> PyResult<Ar
     }
 }
 
+/// Quaternion-based collect of two parallel runs of 1q gates.
+#[derive(Clone, Debug)]
+struct Separable1q {
+    phase: f64,
+    qubits: [VersorSU2; 2],
+}
+impl Separable1q {
+    /// Construct an initial state from a single qubit operation.
+    #[inline]
+    fn from_qubit(n: usize, versor: VersorU2) -> Self {
+        let mut qubits: [VersorSU2; 2] = Default::default();
+        qubits[n] = versor.su2;
+        Self {
+            phase: versor.phase,
+            qubits,
+        }
+    }
+
+    /// Apply a new gate to one of the two qubits.
+    #[inline]
+    fn apply_on_qubit(&mut self, n: usize, versor: &VersorU2) {
+        self.phase += versor.phase;
+        self.qubits[n] = versor.su2 * self.qubits[n];
+    }
+
+    /// Construct the two-qubit gate matrix implied by these two runs.
+    #[inline]
+    fn matrix_into<'a>(&self, target: &'a mut [[Complex64; 4]; 4]) -> ArrayView2<'a, Complex64> {
+        let q0 = VersorU2 {
+            phase: self.phase,
+            su2: self.qubits[0],
+        }
+        .matrix_contiguous();
+
+        // This is the manually unrolled Kronecker product.
+        let [iz, iy, ix, scalar] = *self.qubits[1].0.quaternion().coords.as_ref();
+        let q1_row = [Complex64::new(scalar, iz), Complex64::new(iy, ix)];
+        for out_row in 0..2 {
+            for out_col in 0..4 {
+                target[out_row][out_col] = q1_row[(out_col & 2) >> 1] * q0[out_row][out_col & 1];
+            }
+        }
+        let q1_row = [Complex64::new(-iy, ix), Complex64::new(scalar, -iz)];
+        for out_row in 0..2 {
+            for out_col in 0..4 {
+                target[out_row + 2][out_col] =
+                    q1_row[(out_col & 2) >> 1] * q0[out_row][out_col & 1];
+            }
+        }
```

Comment:
Is there a nalgebra method to do the kronecker product like this? I'm guessing not because we're not building the matrix representation for q1. But figured there might be a small chance.

Reply:
I think I did it like this because I didn't write out the 2x2 matrix of one of the quaternions while doing it (only a 2x1 row twice).  I don't really know _why_ I did that - probably faster to write out the two matrices and use `kron`.

Reply:
There's no `kronecker_to` method in `nalgebra`, which is I think what I needed here.  For the time being, I've just left this as-is along with all the other 2q inefficiencies, which I think are best left for follow-ups.

Reply:
Maybe something to open an issue or contribute in the future then.